iothub/device/src/Transport/HttpClientHelper.cs

```diff
@@ -63,51 +62,50 @@ IWebProxy proxy
 #if NET451
             TlsVersions.Instance.SetLegacyAcceptableVersions();
 
-            WebRequestHandler handler = httpClientHandler as WebRequestHandler;
+            _httpClientHandler = httpClientHandler;
+
             if (clientCert != null)
             {
-                if (handler == null)
+                if (_httpClientHandler == null)
                 {
-                    handler = new WebRequestHandler();
+                    _httpClientHandler = new WebRequestHandler();
                 }
 
-                handler.ClientCertificates.Add(clientCert);
+                (_httpClientHandler as WebRequestHandler).ClientCertificates.Add(clientCert);
```

Comment:
Why not specify WebRequestHandler as the type on line 41?

Reply:
we use WebRequestHandler only for NET451
for the other frameworks, it's a HttpClientHandler